### PR TITLE
Upgrade nixpkgs and add Postgres 13 support

### PR DESCRIPTION
**Pull request by monacoremo**
- upgrade nixpkgs to current master
- link against libpq from Postgres 13 by default
- integrate tests against Postgres 13
- Fix tests for a changed error message in Postgres 13

### CODE REVIEW

**Comment by wolfgangwalther** [comment]
There's one more reference to pg12 in `tests/docker-compose.yml` - that could be updated to pg13 as well, I guess. If somebody used it, it'd make sense to use the latest version.

**Comment by monacoremo** [comment]
> There's one more reference to pg12 in `tests/docker-compose.yml` - that could be updated to pg13 as well,

Good point @wolfgangwalther , updated! I used the `13` tag in favor of `latest` or `13.0`, as it should stay resonably reproducible for local testing while not requiring frequent updates.

**Comment by steve-chavez** [comment]
@monacoremo Looks good!

>  weird that the 'io tests' failed this time

Ah, the `Received "segmentation fault" signal` error has happened to me before. Not sure why it does.. I restart manually on those cases :)

File: nix/patches/nixpkgs-revert-ghc-bootstrap.patch [hunk 1]
```diff
@@ -52,29 +53,29 @@ index 41af279e83f..5853a470efc 100644
...
-
+ 
 -    ghc865Binary = callPackage ../development/compilers/ghc/8.6.5-binary.nix { };
 +    ghc863Binary = callPackage ../development/compilers/ghc/8.6.3-binary.nix { };
-
-     ghc865 = callPackage ../development/compilers/ghc/8.6.5.nix {
-       bootPkgs = packages.ghc822Binary;
-@@ -56,31 +57,25 @@ in {
+ 
+     ghc8102Binary = callPackage ../development/compilers/ghc/8.10.2-binary.nix {
```
Comment (steve-chavez) [inline nix/patches/nixpkgs-revert-ghc-bootstrap.patch:76]: Currently we're using `ghc883`:
https://github.com/PostgREST/postgrest/blob/d395bb60522aa228a534d9181339519f171fe6b2/default.nix#L5-L6

Can I switch that to `ghc8102Binary` to use GHC 8.10 and expect to all work the same?
GHC 8.10 might contain some memory usage improvements and I was thinking to benchmark it :)

Comment (steve-chavez) [reply nix/patches/nixpkgs-revert-ghc-bootstrap.patch:76]: Oh, actually I've just tried that and I get a nix failure:
```
error: attribute 'haskellCompilerName' missing, at /nix/store/yacxa1kgxalkyw7fyk27xm76vp7gjqwd-source/pkgs/development/haskell-modules/make-package-set.nix:136:32
```

No problem though. That was more of a question.

Comment (monacoremo) [reply nix/patches/nixpkgs-revert-ghc-bootstrap.patch:76]: @steve-chavez it probably works for normal builds, but for the fully static build we need `static-haskell-nix` to catch up: https://github.com/nh2/static-haskell-nix/pull/98

**Pull request merged**